test/models/book_test.rb

```diff
@@ -83,4 +83,87 @@ class BookTest < ActiveSupport::TestCase
     assert_not_nil Book.find_by_obj_id(123)
     assert_not_nil Book.find_by_obj_id(234)
   end
+
+  test 'internet_archive_url returns correct url based on IA identifier' do
+    b1 = books(:one) #fixture data
+    assert_equal "https://archive.org/details/#{b1.ia_identifier}", b1.internet_archive_url
+
+  end
+
+  test 'hathitrust_handle works only if book exists in hathitrust' do 
+    b1 = books(:one)
+    b4 = books(:four)
+
+    assert !b1.exists_in_hathitrust
+    assert_equal "", b1.hathitrust_handle
+
+    assert b4.exists_in_hathitrust
+    assert_equal "https://hdl.handle.net/2027/uiuc.#{b4.obj_id}", b4.hathitrust_handle
+  end
+
+  test 'params_from_marcxml_record extracts individual data and returns hash' do 
```

Comment:
This test isn't actually invoking `Book.params_from_marcxml_record()`.

Reply:
I'm stuck on invoking this method successfully in the test. I set it up to parse the data using:
```
b5 = books(:five)
record = Nokogiri::XML(b5.raw_marcxml).root 
key = b5.source_path
```
When I call on `b5.params_from_marcxml_record()` I get an error saying undefined method. Happy to hop on a call to figure this out, but I will skip this test for now.

Reply:
`params_from_marcxml_record()` is a class method, meaning that it's not bound to an instance, but to the Book class itself. Class methods are defined like `def self.method_name`, as opposed to `def method_name` which are instance methods.

Class methods must be called on their class: `Book.params_from_marcxml_record()`

Reply:
Thank you! I completely missed that this was a Class method. The test is now updated and invokes the Class method.